source/darwin/darwin_pki_utils.c

```diff
@@ -104,6 +104,64 @@ int aws_import_public_and_private_keys_to_identity(
         goto done;
     }
 
+    /*
+     * If the key format is unknown, we tried to decode the key into DER format import it.
+     * The PEM file might contians multiple key sections, we will only add the first succeed key into the keychain.
+     */
+    if (key_status == errSecUnknownFormat) {
```

Comment:
nit: can you pull this into a local helper? this function is already pretty big and error handling will be easier